### PR TITLE
Add login page

### PR DESCRIPTION
**Pull request by roshni73**
## Addresses:
- https://github.com/IFRCGo/alert-hub-web-app/issues/113

## Changes
- Add login page

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/Login/index.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                    firstName: selectedUser.first ?? undefined,
+                    lastName: selectedUser.last ?? undefined,
+                    displayName: getUserName({
+                        first_name: selectedUser.first,
+                        last_name: selectedUser.last,
+                        username: selectedUser.username,
+                    }),
+                    token: selectedUser.token,
+                    expires: selectedUser.expires,
+                });
```
Comment (AdityaKhatri) [inline src/views/Login/index.tsx]: This should be done after the request is submitted. Let me know if a call is necessary.

File: src/views/Login/index.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    {
+        id: 2,
+        username: 'janedoe',
+        password: 'password456',
+        first: 'Jane',
+        last: 'Doe',
+        token: 'dummyToken456',
+        expires: '2025-01-31T23:59:59Z',
+    },
+];
```
Comment (AdityaKhatri) [inline src/views/Login/index.tsx]: We don't need 2 users.

File: src/utils/domain/user.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import {
+    isNotDefined,
+    isTruthyString,
+} from '@togglecorp/fujs';
+
+interface User {
+    first_name: string | undefined | null;
+    last_name: string | undefined | null;
+    username: string;
+}
```
Comment (AdityaKhatri) [inline src/utils/domain/user.ts:10]: I think we can take it from server. If the server doesn't have the type right now, add a TODO here.

File: src/views/Login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+const usersData = [
+    {
+        id: 1,
+        username: 'johndoe',
+        password: 'password123',
+        first: 'John',
+        last: 'Doe',
+        token: 'dummyToken123',
+        expires: '2024-12-31T23:59:59Z',
```
Comment (AdityaKhatri) [inline src/views/Login/index.tsx]: This should follow the user type defined in domain

File: src/views/Login/index.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                    expires: selectedUser.expires,
+                });
+            },
+        ),
+        [validate, setError, setUser],
+    );
+
+    const handleCaptchaVerify = (token: string) => {
+        setCaptchaToken(token);
+    };
```
Comment (AdityaKhatri) [inline src/views/Login/index.tsx]: setCaptchaToken can be directly set in the input

File: src/views/Login/index.tsx [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+                        to="login" // Note:Fixme:Loginresendvalidation
+                        title={strings.loginResendValidationTitle}
+                        withUnderline
+                    >
+                        {strings.loginResendValidation}
+                    </Link>
+                </div>
+                <div className={styles.actions}>
+                    <HCaptcha
+                        sitekey="10000000-ffff-ffff-ffff-000000000001"
```
Comment (AdityaKhatri) [inline src/views/Login/index.tsx]: fetch this from environment.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: src/components/Navbar/index.tsx [hunk 7]
```diff
@@ -60,13 +59,19 @@ function Navbar(props: Props) {
...
+                        {strings.appLogin}
+                    </Link>
+                    {/* <Button
                         name={undefined}
                         variant="primary"
                         onClick={undefined}
                     >
                         {strings.appLogin}
-                    </Button>
+                    </Button> */}
```
Comment (barshathakuri) [inline src/components/Navbar/index.tsx]: Please remove unwanted comments.

File: src/views/Login/i18n.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
+{
+    "namespace": "login",
+    "strings": {
+        "loginTitle":"IFRC GO - Login",
+        "loginHeader":"Login",
+        "loginSubHeader":"If you are staff, member or volunteer of the Red Cross Red Crescent Movement (National Societies, the IFRC and the ICRC) login with you email and password.",
+        "loginEmailUsername":"Email/Username",
```
Comment (barshathakuri) [inline src/views/Login/i18n.json]: Use only Email
```suggestion
        "loginEmailUsername":"Email/Username",
```

File: src/views/Login/i18n.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+        "loginPassword":"Password",
+        "loginRecoverTitle":"Recover password",
+        "loginShowUsernameTitle":"Show me my username",
+        "loginResendValidation":"Re-send validation email",
+        "loginResendValidationTitle":"I didn't get my validation email",
+        "loginForgotUserPass":"Forgot your password/username?",
```
Comment (barshathakuri) [inline src/views/Login/i18n.json]: We are only using email
```suggestion
        "loginForgotUserPass":"Forgot your password?",
```

File: src/views/Login/index.tsx [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+        first: 'Jane',
+        last: 'Doe',
+        token: 'dummyToken456',
+        expires: '2025-01-31T23:59:59Z',
+    },
+];
+
+const formSchema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        username: {
```
Comment (barshathakuri) [inline src/views/Login/index.tsx]:
```suggestion
        email: {
```

File: src/views/Login/index.tsx [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+
+    const handleFormSubmit = useMemo(
+        () => createSubmitHandler(
+            validate,
+            setError,
+            () => {
+                const selectedUser = usersData[0];
+                setUser({
+                    id: selectedUser.id,
+                    username: selectedUser.username,
```
Comment (barshathakuri) [inline src/views/Login/index.tsx]: Remove all username
```suggestion
                    username: selectedUser.username,
```

**Review by AdityaKhatri — APPROVED** [review]

**Review by barshathakuri — APPROVED** [review]

**Pull request merged**